### PR TITLE
bpo-32549: Compile OpenSSL 1.1.0 on Travis CI

### PR DESCRIPTION
**Pull request by tiran**
Use an improved version of multissl test helper to compile a local copy
of OpenSSL 1.1.0g.

Signed-off-by: Christian Heimes <christian@python.org>

https://bugs.python.org/issue32549

### CODE REVIEW

**Comment by tiran** [comment]
@brettcannon @vstinner

This PR is a prerequisite for several of my ssl improvements. Do you agree with my approach? I'm going to remove the hak ss soon as Travis has OpenSSL 1.0.2+ in a base image.

Also see https://github.com/travis-ci/travis-ci/issues/9069

**Review by brettcannon — APPROVED** [review]
LGTM; whatever works to unblock you is basically fine by me. 😄

**Comment by tiran** [comment]
@brettcannon thanks! I'm also changing travis CI settings from monthly to daily cron job on master. This will populate the cache tonight. Further runs should reuse pre-cached openssl builds.

**Comment by tiran** [comment]
s/change/add/g

We didn't have cronjob runs at all. I misunderstood the UI.

File: .travis.yml [hunk 1]
```diff
@@ -71,6 +88,13 @@ before_script:
         echo "Only docs were updated, stopping build process."
         exit
       fi
+      if [ "${TESTING}" != "docs" ]; then
+        # clang complains about unused-parameter a lot, redirect stderr
```
Comment (pitrou) [inline .travis.yml:92]: At some point, perhaps the `before_script` should be factored out instead of duplicating all its contents?

**Pull request merged**